### ISSUE TITLE
[Bug] Random walk order on visit

### ISSUE THREAD

**Issue by ChrisIsKing** [Jaseci-Labs/jaseci#1350]
## **Describe the bug**

When fetching nodes for graph traversal, the order of the nodes in the returned list is always random each time the program is run. The expected default behavior is for the nodes to be returned in the same order they were attached to the parent node. An optional feature would be to allow depth-first search traversal, and a nice-to-have feature would be the ability to order nodes by a particular node value.

### **To Reproduce**
1. Run the code below multiple times.
2. Observe the order in which nodes are visited each run.

```
node test_node {
    has value: int;
}

walker test_walker {
    has visited_nodes: list = [];

    can visit_node with `root entry {
        visit [-->];
    }

    can log_visit with object exit {
        self.visited_nodes.append(here);
    }
}

with entry {
    for i in range(10) {
        print("Creating node with value: ", i);
        root ++> (next := test_node(value=i));
    }
    wlk_obj = root spawn test_walker();
    print(wlk_obj.visited_nodes);
}
```

### Example Output
Run 1 output:
```
Creating node with value:  0
Creating node with value:  1
Creating node with value:  2
Creating node with value:  3
Creating node with value:  4
Creating node with value:  5
Creating node with value:  6
Creating node with value:  7
Creating node with value:  8
Creating node with value:  9
[Root(), test_node(value=3), test_node(value=7), test_node(value=0), test_node(value=5), test_node(value=6), test_node(value=2), test_node(value=1), test_node(value=8), test_node(value=4), test_node(value=9)]
```

Run 2 output:
```
Creating node with value:  0
Creating node with value:  1
Creating node with value:  2
Creating node with value:  3
Creating node with value:  4
Creating node with value:  5
Creating node with value:  6
Creating node with value:  7
Creating node with value:  8
Creating node with value:  9
[Root(), test_node(value=7), test_node(value=4), test_node(value=8), test_node(value=1), test_node(value=5), test_node(value=9), test_node(value=2), test_node(value=0), test_node(value=6), test_node(value=3)]
```

The nodes should be visited and returned in the order in which they were attached to the root node, for example:
`[Root(), test_node(value=0), test_node(value=1), test_node(value=2), test_node(value=3), test_node(value=4), test_node(value=5), test_node(value=6), test_node(value=7), test_node(value=8), test_node(value=9)]
`

**Comment by marsninja** [comment]
Yes, this is a valid issue we should fix. We should check the implementation of the connect operator function and the walk function to see how these orders get mixed up. Lists are supposed to be ordered.

**Comment by ChrisIsKing** [comment]
Closing issue. Fixed in #1352
